control-admin/src/main/java/fi/nls/oskari/control/admin/LayerAdminUsageCheckHandler.java

```diff
@@ -0,0 +1,64 @@
+package fi.nls.oskari.control.admin;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.*;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.ServiceRuntimeException;
+import fi.nls.oskari.util.ResponseHelper;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.*;
+
+@OskariActionRoute("LayerAdminUsageCheck")
+public class LayerAdminUsageCheckHandler extends RestActionHandler {
+    private static final Logger LOG = LogFactory.getLogger(LayerAdminUsageCheckHandler.class);
+
+    private static final String PARAM_LAYER_ID = "id";
+    private OskariLayerService mapLayerService;
+
+    @Override
+    public void init() {
+        try {
+            mapLayerService = OskariComponentManager.getComponentOfType(OskariLayerService.class);
+        } catch (Exception e) {
+            throw new ServiceRuntimeException("Exception occured while initializing map layer service", e);
+        }
+    }
+
+    @Override
+    public void handleGet(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+        final int layerId = params.getRequiredParamInt(PARAM_LAYER_ID);
+        LOG.info("Checking layer usage in other layers for layerId: ", layerId);
+        Map<String, Set<Integer>> layerUsages = new HashMap<>();
```

Comment:
Change the type here to generic JSONObject since that's when ResponseHelper writes the json-headers: https://github.com/oskariorg/oskari-server/blob/master/service-control/src/main/java/fi/nls/oskari/util/ResponseHelper.java#L32-L35


```suggestion
        JSONObject response = new JSONObject();
```